File: C++/2-Medium/00162._寻找峰值/00162._寻找峰值.cpp
#include <iostream>
#include <vector>
using namespace std;

// 寻找峰值函数
int findPeakElement(vector<int>& nums) {
    int left = 0, right = nums.size() - 1, mid;
    
    // 使用二分查找定位峰值
    while (left < right) {
        mid = left + (right - left) / 2;
        if (nums[mid] > nums[mid + 1])
            right = mid; // 峰值在左侧
        else
            left = mid + 1; // 峰值在右侧
    }
    return left; // left 和 right 相遇，指向峰值
}

int main() {
    // 示例数组
    vector<int> nums = {1, 2, 1, 3, 5, 6, 4};
    cout << "峰值元素的位置: " << findPeakElement(nums) << endl;
    return 0;
}

File: Python/2-Medium/00162._寻找峰值/00162._寻找峰值.py
from typing import List

def find_peak_element(nums: List[int]) -> int:
    """
    使用二分查找算法寻找峰值元素的索引。
    
    Args:
    nums: 一个整数数组。
    
    Returns:
    峰值元素的索引。
    
    示例:
    >>> find_peak_element([1, 2, 3, 1])
    2
    """
    left, right = 0, len(nums) - 1
    while left < right:
        mid = (left + right) // 2
        # 如果中间元素小于其右侧元素，则峰值在右侧
        if nums[mid] < nums[mid + 1]:
            left = mid + 1
        else:
            # 否则，峰值在左侧或就是中间元素本身
            right = mid
    return left

# 运行代码
# 示例数组
nums_example = [1, 2, 3, 1]
# 寻找峰值
peak_index = find_peak_element(nums_example)
print(peak_index)  # 输出: 2